pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StPointFunction.java

```diff
@@ -65,7 +65,7 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
     if (arguments.size() == 3) {
       transformFunction = arguments.get(2);
       Preconditions.checkArgument(transformFunction instanceof LiteralTransformFunction,
-          "Third argument must be a literal of integer: %s", getName());
+          "Third argument must be a literal of boolean: %s", getName());
```

Comment:
ST_POINT is ANY on 3rd argument, should it be boolean or integer?

Reply:
It should be boolean. We allow int for backward compatibility only because this function was introduced before BOOLEAN is supported